solana/ed25519/src/lib.rs

```diff
@@ -266,6 +266,13 @@ fn base64_display(bytes: &[u8], fmtr: &mut fmt::Formatter) -> fmt::Result {
     fmtr.write_str(unsafe { core::str::from_utf8_unchecked(&buf[..len]) })
 }
 
+/// Displays slice using base58 encoding.
+fn base58_display(bytes: &[u8], fmtr: &mut fmt::Formatter) -> fmt::Result {
+    let data = bs58::encode(bytes).into_string();
+    // SAFETY: base58 fills the buffer with ASCII characters only.
```

Comment:
There’s no `unsafe` code so there’s no need for `SAFETY` comment.
Alternative is to actually use some unsafe to optimise things a bit:

```rust
/// Displays slice using base58 encoding.
// TODO(mina86): Get rid of this once bs58 has this feature.  There’s currently
// PR for that: https://github.com/Nullus157/bs58-rs/pull/97
fn base58_display(bytes: &[u8], fmtr: &mut fmt::Formatter) -> fmt::Result {
    debug_assert!(self.0.len() <= 65);
    // The largest buffer we’re ever encoding is 65-byte long.  Base58
    // increases size of the value by less than 40%.  96-byte buffer is
    // therefore enough to fit the largest value we’re ever encoding.
    let mut buf = [0u8; 96];
    let len = bs58::encode(self.0).into(&mut buf[..]).unwrap();
    let output = &buf[..len];
    // SAFETY: We know that alphabet can only include ASCII characters
    // thus our result is an ASCII string.
    fmtr.write_str(unsafe { std::str::from_utf8_unchecked(output) })
}
```

Reply:
oh i see that makes sense. I was wondering why it was the same in `base64_display` method. Got it now.

Reply:
In `base64_display` there was a SAFETY comment because of the `unsafe` block wrapping `core::str::from_utf8_unchecked` call.

Reply:
gotcha